src/components/BulkUploadListItem.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import prettyBytes from 'pretty-bytes';
+import {
+  ArrowPathIcon,
+  CheckCircleIcon,
+  ClockIcon,
+  XCircleIcon,
+} from '@heroicons/react/24/outline';
+import {
+  localizeDateTime,
+  relativizeDateTime,
+} from '../utils/datetime';
+import type { ApiBulkUpload } from '../pdc-api';
+import { EmailLink } from './EmailLink';
+import './BulkUploadListItem.css';
+
+enum BulkUploadStatuses {
+  pending = 'pending',
+  inProgress = 'in_progress',
+  completed = 'completed',
+  canceled = 'canceled',
+  failed = 'failed',
+}
+
+interface BulkUploadListItemProps {
+  upload: ApiBulkUpload;
+}
+
+export const BulkUploadListItem = ({ upload }: BulkUploadListItemProps) => {
+  const {
+    id, fileName, fileSize, status, createdAt,
+  } = upload;
+
+  const relativeTimestamp = (
+    <time dateTime={createdAt} title={localizeDateTime(createdAt)}>
+      {relativizeDateTime(createdAt)}
+    </time>
+  );
+
+  const formattedFileSize = fileSize ? (
+    <span title={`${fileSize.toLocaleString()} byte${fileSize !== 1 ? 's' : ''}`}>
+      {prettyBytes(fileSize, { space: false })}
+    </span>
+  ) : null;
+
+  let icon = null;
+  const metadata = [];
+
+  switch (status) {
+    case BulkUploadStatuses.pending:
+      icon = <ClockIcon className="icon" />;
+      metadata.push(relativeTimestamp, formattedFileSize, 'Waiting to process…');
+      break;
+    case BulkUploadStatuses.inProgress:
+      icon = <ArrowPathIcon className="icon rotating" />;
+      metadata.push(relativeTimestamp, formattedFileSize, 'Processing…');
+      break;
+    case BulkUploadStatuses.completed:
+      icon = <CheckCircleIcon className="icon" />;
+      metadata.push(relativeTimestamp, formattedFileSize);
+      break;
+    case BulkUploadStatuses.canceled:
+      icon = <XCircleIcon className="icon" />;
+      metadata.push(
+        (
+          <>
+            Canceled
+            {' '}
+            {relativeTimestamp}
+          </>
+        ),
+        formattedFileSize
+      );
+      break;
+    case BulkUploadStatuses.failed:
+      icon = <XCircleIcon className="icon" />;
+      metadata.push(
+        (
+          <>
+            Failed
+            {' '}
+            {relativeTimestamp}
+          </>
+        ),
+        formattedFileSize,
+        (
+          <EmailLink
+            to="info@philanthropydatacommons.org"
+            subject={`Bulk upload error (ID: ${id})`}
+          >
+            Email the dev team for help
+          </EmailLink>
+        )
+      );
+      break;
+    default:
+  }
+
+  // We delimit metadata elements with a bullet padded by two spaces.
```

Comment:
this raises a deep philosophical question: if something has one space on either side of it, is it padded by two spaces or is it padded by one space.  If two: can something be padded by one space?  If one: are we saying the spaces are the same space?

Reply:
Superficially, you're right, and I can illustrate it with a CSS example: we'd never say "padded by 20px" to mean the sum of padding. We would mean 20px in each relevant dimension. E.g., a box with 5px of padding on each side would be "padded by 5px", not padded by 20px. (If it were padded in only one dimension, say, horizontally but not vertically, then either context or explanation would make that clear.)

However, this is a _bullet_, which has only one external surface. Therefore to describe its padding, we _do_ sum the values of "both sides" — that is, spacing off any axis. And so one space on either "side" is really two spaces distributed symmetrically off the horizontal poles of the unisurface. I thought it easier to say "padded by two spaces".

Thank you for coming to my TED talk.

Reply:
That was offensively clear. Or rather, so clear that your explanation is offensive to me.